listen.py

```diff
@@ -43,6 +43,41 @@ def from_json(cls, j):
             data=j.get('track_metadata')
         )
 
+    @classmethod
+    def from_influx(cls, row):
+        """ Factory to make Listen objects from an influx row
+        """
+        dt = datetime.strptime(row['time'] , "%Y-%m-%dT%H:%M:%SZ")
+        t = int(dt.strftime('%s'))
+        mbids = []
+        if row.get('artist_mbids'):
```

Comment:
Ah, I didn't realise that the problem was that the data was coming back as `None` from influx. Why is this? Is it because we fill out these fields on insert? or because there is a schema defined which always fills in these fields?
If the fields are optional, we shouldn't add them on insert if they have no data. If you need to check if they're None, use the first suggestion that I gave, where we move the data into a separate variable